browser/ui/views/tabs/tab_drag_controller.cc

```diff
@@ -187,9 +187,9 @@ void TabDragController::DetachAndAttachToNewContext(
     std::vector<tabs::TabHandle> tabs;
     auto* tab_strip_model = browser->tab_strip_model();
     DCHECK_EQ(tab_strip_model, attached_context_->GetTabStripModel());
-    for (const auto& tab_drag_data : drag_data_) {
+    for (size_t i = first_tab_index(); i < drag_data_.size(); ++i) {
```

Comment:
Nice- took me a minute to spot the fix 😄 Basically starting at `first_tab_index` and not iterating every item

Reply:
Please consider changing this to something like:

```cxx
auto drag_data = base::span(drag_data_).subspan(first_tab_index());
for (const auto& tab_drag_data : drag_data) {
  ...
```

This will avoid build failures with the unsafe buffers warning as we don't access the subscript directly.

Reply:
Oops - I missed this before merging; @Ilie-Lesan you're welcome to do a follow up or I can do that. Just let me know!

Reply:
Pushed https://github.com/brave/brave-core/pull/26055